File: deprecated_versions/deprecated_pysfml_version/sbfury/control.py
from PySFML import sf

class Control:

    def __init__(self, input):
        self.input = input
        self._create_map()
        self.update()

    def _create_map(self):
        self.map = {
                'left': sf.Key.Left,
                'right': sf.Key.Right,
                'up': sf.Key.Up,
                'down': sf.Key.Down,
                'attack': sf.Key.D,
                'jump': sf.Key.A,
                'run': sf.Key.S,
                'special': sf.Key.Space,
                }

    def update(self):

        for k, v in self.map.items():
            if self.input.IsKeyDown(v):
                setattr(self, k, True)
            else:
                setattr(self, k, False)

File: deprecated_versions/deprecated_cpp_version/control.cc
/*
 * Shaolin's Blind Fury
 *
 * Copyright 2008, 2009 - Hugo Ruscitti
 * License: GPLv3 (see http://www.gnu.org/licenses/gpl.html)
 */

#include "SDL/SDL.h"
#include "control.h"

Control :: Control()
{
    up = false;
    down = false;
    left = false;
    right = false;
    attack = false;
    jump = false;

    move = false;
}

void Control :: update(float dt)
{
    key = SDL_GetKeyState(NULL);

    up = (key[SDLK_UP] || key[SDLK_k]);
    down = (key[SDLK_DOWN] || key[SDLK_j]);
    left = (key[SDLK_LEFT] || key[SDLK_h]);
    right = (key[SDLK_RIGHT] || key[SDLK_l]);

    move = (up || down || left || right);

    attack = (key[SDLK_s]);
    jump = (key[SDLK_a]);
}
